bundles/org.openhab.core.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnection.java

```diff
@@ -561,14 +560,21 @@ public boolean hasSubscribers() {
             return future;
         }
         if (client.getState().isConnected()) {
-            client.subscribe(topic, qos, clientCallback).whenComplete((s, t) -> {
-                if (t == null) {
-                    logger.trace("Subscribed {} to topic {}", subscriber, topic);
-                    future.complete(true);
-                } else {
-                    future.completeExceptionally(new MqttException(t));
-                }
-            });
+            if (!client.isSubscribed(topic)) {
+                // "real" subscription for first subscriber
+                client.subscribe(topic, qos, clientCallback).whenComplete((s, t) -> {
```

Comment:
This is not threadsafe, or is it?
Two threads could both do the check for isSubscribed and then do the subscription.
The checking and the adding to the list need to be atomic.